pymc/distributions/continuous.py

```diff
@@ -2628,15 +2628,18 @@ def dist(cls, nu=1, sigma=None, lam=None, sd=None, *args, **kwargs):
         lam, sigma = get_tau_sigma(lam, sigma)
         sigma = at.as_tensor_variable(sigma)
 
-        # mode = at.as_tensor_variable(0)
-        # median = at.as_tensor_variable(sigma)
-        # sd = at.as_tensor_variable(sigma)
-
         assert_negative_support(nu, "nu", "HalfStudentT")
         assert_negative_support(lam, "lam", "HalfStudentT")
         assert_negative_support(sigma, "sigma", "HalfStudentT")
 
         return super().dist([nu, sigma], *args, **kwargs)
+    
+    def get_moment(rv, size, nu, sigma):
+        lam, sigma = get_tau_sigma(None, sigma)
```

Comment:
This line is not necessary as sigma is already given

Reply:
And you need to make sure sigma is broadcasted to the shape of nu, with `sigma, _ = at.broadcast_arrays(sigma, nu)` Otherwise you might return the wrong shape for the moment (e.g., when `sigma=0` and `nu=[1,2,3,4,5]`)

Reply:
Thanks for the tip, change applied.